resources/type-system/inference.md

```diff
@@ -286,9 +291,9 @@ type schema `S` as follows:
     `Stream<S1>` for some `S1`, then the context type is `S1`.
   - If the function expression is declared `sync*` and `S` is of the form
     `Iterable<S1>` for some `S1`, then the context type is `S1`.
```

Comment:
If the imposed context type schema is `Stream<int>?` or `Object` (or a number of other types that aren't of the form `Stream<S1>` for any `S1`) then we do not get a binding of `S1` here.

We already have the notion of 'the element type of a generator', so we'd need 'the element type of a type'. This would be `S` for any type of the form `Stream<S>`, `Stream<S>?`, `Iterable<S>`, `Iterable<S>?`, and it would be `Object?` for any other type. A generator function is a compile-time error unless it has a return type which is a supertype of `Iterable<Never>` resp. `Stream<Never>`, but the context type could be anything, and it could also be `dynamic` or `void` or `Stream<int>?` and so on.

This is going to give us a better context type for generators.